cocotb_test/run.py

```diff
@@ -243,50 +247,62 @@ def _run_icarus(toplevel, libs_dir, verilog_sources_abs, sim_build_dir, include_
 def _run_questa(toplevel, libs_dir, verilog_sources_abs, vhdl_sources_abs, sim_build_dir, ext_name, toplevel_lang, include_dir_abs):
 
     inculde_dir_cmd = []
-    for dir in include_dir_abs: 
+    for dir in include_dir_abs:
         inculde_dir_cmd.append('+incdir+' + dir)
 
     do_script = '''# Autogenerated file
     onerror {
         quit -f -code 1
     }
     '''
-    
+
     if vhdl_sources_abs:
-        do_script += 'vcom -mixedsvvh +define+COCOTB_SIM {INCDIR} {VHDL_SOURCES}\n'.format(VHDL_SOURCES=" ".join(vhdl_sources_abs), INCDIR=" ".join(inculde_dir_cmd))
+        do_script += 'vcom -mixedsvvh +define+COCOTB_SIM {INCDIR} {VHDL_SOURCES}\n'.format(
+            VHDL_SOURCES=as_tcl_value(vhdl_sources_abs),
+            INCDIR=as_tcl_value(inculde_dir_cmd)
+        )
         os.environ["GPI_EXTRA"] = 'fli'
-         
+
     if verilog_sources_abs:
-        do_script += 'vlog -mixedsvvh +define+COCOTB_SIM -sv {INCDIR} {VERILOG_SOURCES}\n'.format(VERILOG_SOURCES=" ".join(verilog_sources_abs), INCDIR=" ".join(inculde_dir_cmd))
-    
+        do_script += 'vlog -mixedsvvh +define+COCOTB_SIM -sv {INCDIR} {VERILOG_SOURCES}\n'.format(
+            VERILOG_SOURCES=as_tcl_value(verilog_sources_abs),
+            INCDIR=as_tcl_value(inculde_dir_cmd)
```

Comment:
Ah crap, these were wrong - they should have been `" ".join(as_tcl_value(v) for v in verilog_sources_abs)`